### PR TITLE
Consolidate common folders to avoid hitting the FSEvent limit

### PR DESCRIPTION
**Pull request by ashaffer**
@es128  Here's the PR for #482. Let me know if you want me to fix anything up or change it.

It does a great job on my project. Goes from 200+ instances to 19-35 (it's somewhat non-deterministic because the way the algorithm works, it depends on the order they accumulate in).

### CODE REVIEW

File: lib/fsevents-handler.js [hunk 1]
```diff
@@ -89,6 +98,33 @@ function setFSEventsListener(path, realPath, listener, rawEmitter) {
   };
 }
 
+var consolidateThreshhold = 10;
+
+// Decide whether or not we should start a new higher-level
+// parent watcher
+function couldConsolidate(path) {
+  var candidates = {};
```
Comment (paulmillr) [inline lib/fsevents-handler.js]: Could you conform to our code style here please?

Comment (ashaffer) [inline lib/fsevents-handler.js]: Do you want the threshold variable at the top?

Also I noticed `candidates` is no longer being used.

**Comment by ashaffer** [comment]
Fixed style.

File: lib/fsevents-handler.js [hunk 2]
```diff
@@ -89,6 +102,30 @@ function setFSEventsListener(path, realPath, listener, rawEmitter) {
...
+        return true;
+      }
+    }
+  }
+
+  return false;
+}
+
+// Find the parent of a given path
+function parentOf (path) {
```
Comment (paulmillr) [inline lib/fsevents-handler.js]: sysPath.dirname`??

Comment (ashaffer) [inline lib/fsevents-handler.js]: Fixed. Sorry, didn't realize that `dirname` worked for directories parents as well.

**Comment by ashaffer** [comment]
Any chance we could get a patch version bump for this so I can start using it in watchify without maintaining a fork?

**Pull request merged**